app/redux/selectors/SetsSelectors.js

```diff
@@ -371,23 +372,53 @@ export const getRevision = (state) => stateRoot(state).revision;
 
 // 1rm
 // Get Weights and Velocity for the exercise
-export const getExerciseData = (state, exercise) => {   
+export const getExerciseData = (state, exercise, type) => {   
     const sets = getAllSets(state);
     let data = [];
 
-    sets.forEach((set) => {
-        const repData = set.reps[0].data;
+    const days = AnalysisSelectors.getAnalysisDays(state);
+    const date = new Date();
+    const minDate = new Date(new Date().setDate(date.getDate() - days));
 
-        if (set.exercise === exercise) {
-            data.push([set.weight, Number(RepDataMap.averageVelocity(set.reps[0].data))]);
-        }       
-    });
+    if (type === 'regression') {
+        sets.forEach((set) => {
+            if (set.exercise === exercise && set.reps.length > 0 && set.weight && new Date(set.initialStartTime) >= minDate) {
+                data.push([set.weight, Number(RepDataMap.averageVelocity(set.reps[0].data))]);
+            }       
+        });
+    } else {
+        sets.forEach((set) => {
+            if (set.exercise === exercise && set.reps.length > 0 && set.weight && set.initialStartTime >= minDate) {
+                data.push({ title: set.setID, weight: set.weight, velocity: Number(RepDataMap.averageVelocity(set.reps[0].data)) });
+            }       
+        });
```

Comment:
- [ ] Why does one of the if statements use the date, and the other use just the start time

Reply:
typo